hw2/03-count.js

```diff
@@ -0,0 +1,11 @@
+// Add your code here
+let paragraphInfo = document.getElementById("paraInfo");
+let inputText = document.getElementById("userinput");
+inputText.addEventListener("input", (event) => {
+  const searchText = event.target.value;
+  const regex = new RegExp(searchText, "g");
+  let text = paragraphInfo.innerHTML;
+  text = text.replace(/(<mark style="background-color: yellow;">|<\/mark>)/gim, "");
+  const newText = text.replace(regex, `<mark style="background-color: yellow;">$&</mark>`);
+  paragraphInfo.innerHTML = newText;
+});
```

Comment:
![rutcount3](https://github.com/rutujapadgilwar/frontend_homework/assets/38127011/b918bdb4-a700-4f7c-84da-1a4501ac11fd)
![rutcount2](https://github.com/rutujapadgilwar/frontend_homework/assets/38127011/eb31f88c-7621-461c-a3ee-ee56ee88056c)
![rutcount1](https://github.com/rutujapadgilwar/frontend_homework/assets/38127011/070ef7c8-6fab-451c-a31b-0bb44cc21682)


There are numerous bugs in the implementation of this count exercise. Take a look at the screenshots. These bugs mainly stem from the fact that it does not perform any validity check on the value passed in by the user.